modules/internal/ChapelHashtable.chpl

```diff
@@ -87,8 +87,15 @@ module ChapelHashtable {
           c_memset(ptrTo(ret[slot]), 0:uint(8), sizeofElement);
         }
       }
+      when ArrayInit.gpuInit {
+        // This should match the 'these' iterator in terms of idx->task
+        extern proc chpl_gpu_memset(addr, byte, numBytes);
+        foreach slot in _allSlots(size) {
+          chpl_gpu_memset(ptrTo(ret[slot]), 0:uint(8), sizeofElement);
+        }
+      }
       otherwise {
-        halt("ArrayInit.heuristicInit should have been made concrete");
+        halt("ArrayInit.", initMethod, " should have been made concrete");
```

Comment:
I'm not sure I really care about this, this halt is something that would get triggered because of a bug on our end right and the most important thing is some kind of error gets triggered to let us know to look here. But...

I'm guessing you updated this because it got triggered due to `ArrayInit.gpuInit` was missing.  But "ArrayInit.gpuInit should have been made concrete" still doesn't make sense. I think word "abstract" here was referring to the idea that "heuristicInit" is "abstract" in the sense that it could resolve to several different kinds of init.  But with `Array.gpuInit`

So it seems like the code should have been something like this:

```chpl
otherwise {
  assert(initMethod == ArrayInit.heuristicInit)
  halt("ArrayInit.heuristicInit should have been made concrete");
}
```

Then the error message makes sense for the case it was intended for and we'd still hit an assertion if we add some new concrete kind of "ArrayInit" that we left unhandled.

Reply:
Offline, we decided to reword the error message to "should have been implemented". Note that this is not user-facing. We'll hit this only if we do something fundamentally wrong during development (I did while working on this branch)